helmtestserver/server.go

```diff
@@ -69,9 +75,56 @@ func (s *HelmServer) PackageChart(path string) error {
 // with the given version, to be served by the HelmServer. It returns an
 // error in case of a packaging failure.
 func (s *HelmServer) PackageChartWithVersion(path, version string) error {
+	return s.packageChart(path, version, "")
+}
+
+// PackageSignedChartWithVersion attempts to package the chart at the given path
+// with the given version and sign it using a PGP keyring, to be served by the HelmServer.
+// It returns an error in case of a packaging failure.
+func (s *HelmServer) PackageSignedChartWithVersion(path, version, publicKeyPath string) error {
+	return s.packageChart(path, version, publicKeyPath)
+}
+
+func (s *HelmServer) packageChart(path, version, publicKeyPath string) error {
 	pkg := action.NewPackage()
-	pkg.Destination = s.HTTPServer.Root()
+	pkg.Destination = s.Root()
 	pkg.Version = version
+	if publicKeyPath != "" {
+		secretKeyPath := fmt.Sprintf("%s/%s", s.Root(), "secret.pgp")
```

Comment:
Would be better to have some randomness here to avoid multiple packaging processes from stepping over one another when a single test server is shared.

Reply:
Maybe also consider adding a `defer` to just delete the secret after use, since it's not needed at all later.